novawallet/Modules/BackupAttention/BackupAttentionViewLayout.swift

```diff
@@ -10,4 +38,126 @@ final class BackupAttentionViewLayout: UIView {
     required init?(coder _: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-}
\ No newline at end of file
+
+    override func safeAreaInsetsDidChange() {
+        super.safeAreaInsetsDidChange()
+
+        setupLayout()
+    }
+
+    func bind(viewModel: Model) {
+        if viewModel.button != self.viewModel?.button {
+            let agreeButton = makeAgreeButton(for: viewModel.button)
+            setupAgreeButton(with: agreeButton)
+        }
+
+        self.viewModel = viewModel
+        checkBoxScrollableView.bind(viewModel: viewModel.rows)
+    }
+}
+
+// MARK: Model
+
+extension BackupAttentionViewLayout {
+    struct Model {
+        let rows: CheckBoxIconDetailsScrollableView.Model
+        let button: ButtonModel
+    }
+
+    enum ButtonModel: Equatable {
+        case active(title: String)
+        case inactive(title: String)
+
+        static func == (lhs: ButtonModel, rhs: ButtonModel) -> Bool {
+            switch (lhs, rhs) {
+            case (.active, .active):
+                return true
+            case (.inactive, .inactive):
+                return true
+            default:
+                return false
+            }
+        }
+    }
+}
+
+// MARK: Private
+
+private extension BackupAttentionViewLayout {
+    func setupLayout() {
+        addSubview(checkBoxScrollableView)
+        checkBoxScrollableView.containerView.scrollContentBottomOffset = UIConstants.scrollBottomOffset(for: self)
+        checkBoxScrollableView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+
+        addSubview(blurredBottomView)
+        blurredBottomView.snp.makeConstraints { make in
+            make.height.equalTo(UIConstants.bottomBlurViewHeight(for: self))
+
+            // hiding the border at the edges of the screen
+            make.leading.equalToSuperview().offset(-0.5)
+            make.trailing.bottom.equalToSuperview().offset(0.5)
+        }
+    }
+
+    func makeAgreeButton(for viewModel: ButtonModel) -> UIControl {
+        switch viewModel {
+        case let .active(title):
+            let button = TriangularedButton()
+            button.imageWithTitleView?.title = title
+            button.applyEnabledStyle()
+
+            return button
+        case let .inactive(title):
+            let button = TriangularedBlurButton()
+            button.imageWithTitleView?.title = title
+            button.applyDisabledStyle()
+
+            return button
+        }
+    }
+
+    func setupAgreeButton(with button: UIControl) {
+        guard let agreeButton else {
+            showButtonWithAnimation(button)
+            return
+        }
+
+        changeButtonWithAnimation(button)
+    }
+
+    func changeButtonWithAnimation(_ button: UIControl) {
```

Comment:
We don't need an animation to switch between button's state. Let's use static button and change its title and style to switch between states. For styling we usually use applyEnabledStyle/applyDisableStyle